pkg/sdk/testint/security_integrations_gen_integration_test.go

```diff
@@ -0,0 +1,380 @@
+		id := testClientHelper().Ids.RandomAccountObjectIdentifier()
+		createSAML2Integration(t, id, func(r *sdk.CreateSAML2SecurityIntegrationRequest) {
+			r.WithAllowedEmailPatterns([]sdk.EmailPattern{{Pattern: "^(.+dev)@example.com$"}}).
+				WithAllowedUserDomains([]sdk.UserDomain{{Domain: "example.com"}}).
+				WithComment(sdk.Pointer("a")).
+				WithSaml2EnableSpInitiated(sdk.Pointer(true)).
+				WithSaml2ForceAuthn(sdk.Pointer(true)).
+				WithSaml2PostLogoutRedirectUrl(sdk.Pointer("http://example.com/logout")).
+				WithSaml2RequestedNameidFormat(sdk.Pointer("urn:oasis:names:tc:SAML:1.1:nameid-format:unspecified")).
+				WithSaml2SignRequest(sdk.Pointer(true)).
+				WithSaml2SnowflakeAcsUrl(&acsURL).
+				WithSaml2SnowflakeIssuerUrl(&issuerURL).
+				WithSaml2SpInitiatedLoginPageLabel(sdk.Pointer("label"))
+		})
+		details, err := client.SecurityIntegrations.Describe(ctx, id)
+		require.NoError(t, err)
+
+		assertSAML2Describe(details, saml2details{
+			provider:                  "Custom",
+			enableSPInitiated:         "true",
+			spInitiatedLoginPageLabel: "label",
+			ssoURL:                    "https://example.com",
+			issuer:                    "test",
+			requestedNameIDFormat:     "urn:oasis:names:tc:SAML:1.1:nameid-format:unspecified",
+			forceAuthn:                "true",
+			postLogoutRedirectUrl:     "http://example.com/logout",
+			signrequest:               "true",
+			comment:                   "a",
+			snowflakeIssuerURL:        issuerURL,
+			snowflakeAcsURL:           acsURL,
+			allowedUserDomains:        "[example.com]",
+			allowedEmailPatterns:      "[^(.+dev)@example.com$]",
+		})
+
+		si, err := client.SecurityIntegrations.ShowByID(ctx, id)
+		require.NoError(t, err)
+		assertSecurityIntegration(t, si, id, "SAML2", false, "a")
+	})
+
+	t.Run("CreateSCIM", func(t *testing.T) {
+		networkPolicy, networkPolicyCleanup := testClientHelper().NetworkPolicy.CreateNetworkPolicy(t)
+		t.Cleanup(networkPolicyCleanup)
+
+		id := testClientHelper().Ids.RandomAccountObjectIdentifier()
+		createSCIMIntegration(t, id, func(r *sdk.CreateSCIMSecurityIntegrationRequest) {
+			r.WithComment(sdk.Pointer("a")).
+				WithNetworkPolicy(sdk.Pointer(sdk.NewAccountObjectIdentifier(networkPolicy.Name))).
+				WithSyncPassword(sdk.Pointer(false))
+		})
+		details, err := client.SecurityIntegrations.Describe(ctx, id)
+		require.NoError(t, err)
+
+		assertSCIMDescribe(details, "false", networkPolicy.Name, "GENERIC_SCIM_PROVISIONER", "false", "a")
+
+		si, err := client.SecurityIntegrations.ShowByID(ctx, id)
+		require.NoError(t, err)
+		assertSecurityIntegration(t, si, id, "SCIM - GENERIC", false, "a")
+	})
+
+	t.Run("AlterSAML2Integration", func(t *testing.T) {
+		id := testClientHelper().Ids.RandomAccountObjectIdentifier()
+		createSAML2Integration(t, id, nil)
+
+		setRequest := sdk.NewAlterSAML2IntegrationSecurityIntegrationRequest(id).
+			WithSet(
+				sdk.NewSAML2IntegrationSetRequest().
+					WithComment(sdk.Pointer("a")).
```

Comment:
could you add other set options? Could be even the same as in create, but it will just confirm every option is defined properly and is possible to set through Snowflake.

Reply:
Done.